### PR TITLE
Ticket5118 generate as given name

### PR DESCRIPTION
**Pull request by JamesKingWork**
### Description of work

Added a facility to give a name to save a generated script with whilst still providing a default.

### Ticket

https://github.com/ISISComputingGroup/IBEX/issues/5118

### Acceptance criteria
- [ ] GIVEN I have valid parameters WHEN I click generate script THEN I am asked what name I want to give my script AND this name has a default value of the name we currently generate
    - Ensure the timestamp on the generated name only goes down to the accuracy of seconds
- [ ] GIVEN I have entered a duplicate name for a script WHEN I click to save the script THEN I am asked whether I want to overwrite the file

### Unit tests

- Mostly GUI work so difficult to test, system tests should be done as part of https://github.com/ISISComputingGroup/IBEX/issues/5090 to make up for this

### System tests

- Should be done as part of https://github.com/ISISComputingGroup/IBEX/issues/5090

### Documentation

- Being updated: https://github.com/ISISComputingGroup/ibex_user_manual/wiki/Using-the-Script-Generator

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.scriptgenerator.tests/src/uk/ac/stfc/isis/ibex/scriptgenerator/tests/ScriptGeneratorSingletonTest.java [hunk 1]
```diff
@@ -156,53 +153,4 @@ public void test_GIVEN_new_file_WHEN_generate_THEN_filename_without_version() {
 		}
 	}
 	
-	@Test
```
Comment (Alistair-McGann-Tessella) [inline base/uk.ac.stfc.isis.ibex.scriptgenerator.tests/src/uk/ac/stfc/isis/ibex/scriptgenerator/tests/ScriptGeneratorSingletonTest.java]: Where have these tests been moved to?

File: base/uk.ac.stfc.isis.ibex.ui.widgets/src/uk/ac/stfc/isis/ibex/ui/widgets/SaveScriptGeneratorFileMessageDialog.java [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+import uk.ac.stfc.isis.ibex.logger.IsisLog;
+
+
+/**
+ * A message dialog box that allows you to copy the contents of the dialog and open the file the dialog is tied to.
+ * 
+ * @author https://stackoverflow.com/a/30630475 modified by James King
+ *
+ */
+public class SaveScriptGeneratorFileMessageDialog extends MessageDialog {
```
Comment (Alistair-McGann-Tessella) [inline base/uk.ac.stfc.isis.ibex.ui.widgets/src/uk/ac/stfc/isis/ibex/ui/widgets/SaveScriptGeneratorFileMessageDialog.java]: There is a lot of logic in this class which shouldn't exist in the UI layer/view. Generating the filename, checking files on disc and saving a file are more appropriately placed near the script gen singleton, in the 'model'. Is it possible to move these methods to there? Not having this logic in the UI layer would also make it easier to write/run tests.

File: base/uk.ac.stfc.isis.ibex.scriptgenerator.tests/src/uk/ac/stfc/isis/ibex/scriptgenerator/tests/ScriptGeneratorSingletonTest.java (hunk 1 above)
Comment (JamesKingWork) [reply base/uk.ac.stfc.isis.ibex.scriptgenerator.tests/src/uk/ac/stfc/isis/ibex/scriptgenerator/tests/ScriptGeneratorSingletonTest.java]: I think I had meant to reimplement them somewhere but somehow forgot? Not sure. Anyway they've been added now one of them in here and multiple more in ScriptGeneratorFileHandlerTest.java

File: base/uk.ac.stfc.isis.ibex.ui.widgets/src/uk/ac/stfc/isis/ibex/ui/widgets/SaveScriptGeneratorFileMessageDialog.java (hunk 2 above)
Comment (JamesKingWork) [reply base/uk.ac.stfc.isis.ibex.ui.widgets/src/uk/ac/stfc/isis/ibex/ui/widgets/SaveScriptGeneratorFileMessageDialog.java]: I agree, moved out to ScriptGeneratorFileHandler

**Pull request merged**